model/project_vars.go

```diff
@@ -66,6 +68,61 @@ func FindOneProjectVars(projectId string) (*ProjectVars, error) {
 	return projectVars, nil
 }
 
+// FindMergedProjectVars merges vars from the target project's ProjectVars and its parent repo's vars
+func FindMergedProjectVars(projectID string) (*ProjectVars, error) {
+	project, err := FindOneProjectRef(projectID)
+	if err != nil {
+		return nil, errors.Wrapf(err, "problem getting project '%s'", projectID)
+	}
+	if project == nil {
+		return nil, errors.New(fmt.Sprintf("project '%s' does not exist", projectID))
+	}
+
+	projectVars, err := FindOneProjectVars(projectID)
+	if err != nil {
+		return nil, errors.Wrapf(err, "problem getting project vars for project '%s'", projectID)
+	}
+	if !project.UseRepoSettings {
+		return projectVars, nil
+	}
+
+	repoVars, err := FindOneProjectVars(project.RepoRefId)
+	if err != nil {
+		return nil, errors.Wrapf(err, "problem getting project vars for repo '%s'", project.RepoRefId)
+	}
+	if repoVars != nil {
+		if projectVars == nil {
+			projectVars = repoVars
+			projectVars.Id = project.Id
+		} else {
```

Comment:
To avoid having such nested else statements, you can return after each of the checks above (i.e. if repoVars == nil, return projectVars, and return after line 96 as well)